File: Competition/2019/Golden/Parallel-or-Intersecting-Lines/GraphLine.py
class GraphLine(object):
    """Software class to represent graphical lines."""

    def __init__(self, x1, y1, x2, y2):
        """Initialize this graph line."""

        self.x1 = x1
        self.x2 = x2
        self.y1 = y1
        self.y2 = y2

        self.slope = self.get_slope()
        self.line_type = self.get_line_type()
    
    def get_slope(self):
        """Get the slope of the line"""

        try:
            self.slope = (self.y2 - self.y1) / (self.x2 - self.x1)
        
        except ZeroDivisionError:
            self.slope = None
        
        return self.slope
    
    def get_line_type(self):
        """Get the line type"""

        slope = self.get_slope()

        if slope == None:
            self.line_type = "Vertical"
        
        elif slope == 0:
            self.line_type = "Horizontal"
        
        else:
            self.line_type = "Inclined"
        
        return self.line_type

File: Competition/2019/Golden/Parallel-or-Intersecting-Lines/GraphLine.java
public class GraphLine {

    private double slope;
    private String lineType;
    private double x1;
    private double x2;
    private double y1;
    private double y2;
    
    public GraphLine(double x1, double y1, double x2, double y2)
    {
        this.x1 = x1;
        this.x2 = x2;
        this.y1 = y1;
        this.y2 = y2;
    }

    public double getSlope()
    {
        double slope;
        
        try
        {
            slope = (this.y2 - this.y1) / (this.x2 - this.x1);
        }

        catch (Exception e)
        {
            slope = Double.MAX_VALUE;
        }

        this.slope = slope;

        return this.slope;
    }

    public String getLineType()
    {
        switch (this.getSlope())
        {
            case Double.MAX_VALUE:
                this.lineType = "Vertical";
            
            case 0.0:
                this.lineType = "Horizontal";
            
            default:
                this.lineType = "Inclined";
        }

        return this.lineType;

    }
}
